lib/task/digitalobject/digitalObjectRegenDerivativesTask.class.php

```diff
@@ -182,12 +182,15 @@ public static function regenerateDerivatives(&$digitalObject)
       $derivative->delete();
     }
 
-    // Delete existing transcripts
-    foreach ($digitalObject->propertys as $property)
+    // Delete existing transcripts if 'keepTranscript' option is not sent or it's false
+    if (!isset($options['keepTranscript']) || !$options['keepTranscript'])
     {
-      if ('transcript' == $property->name)
+      foreach ($digitalObject->propertys as $property)
```

Comment:
`QubitInformationObject` implements `getPropertyByName`, should `QubitDigitalObject` do the same?